pkg/libartifact/store/store.go

```diff
@@ -132,7 +132,14 @@ func (as ArtifactStore) Inspect(ctx context.Context, nameOrDigest string) (*liba
 	if err != nil {
 		return nil, err
 	}
-	inspectData, _, err := artifacts.GetByNameOrDigest(nameOrDigest)
+
+	// Normalize name to :latest if no tag/digest present for single lookup
+	lookupName := nameOrDigest
+	if !strings.Contains(nameOrDigest, ":") && !strings.Contains(nameOrDigest, "@") {
```

Comment:
The test for `podman artifact inspect with full or partial digest` fails because `:latest` is being added. I think we should handle the `:latest` tag inside `artifacts.GetByNameOrDigest` instead.